### PR TITLE
Fix/compile errors in v4.0.0

### PR DESCRIPTION
**Pull request by nikolay-vasilev-prime**
#### Overview

> Closes: #787 and #788 

Fixes non compilable issues with v4.0.0 and adding test and analyze steps to the CI

#### Checklist

*Implementation*
- [x] Implementation matches ticket acceptance criteria and technical notes
- [x] Manually tested against Acceptance Criteria
- [x] UI checked in Light / Dark mode

*Stability*
- [x] Checked if changes affect any features and verified affected features work as expected
- [ ] Added unit tests for new code and verified existing tests work as expected

*Code quality*
- [ ] Updated `CHANGELOG.md`, `README.md` and package versions in `pubspec.yaml`
- [ ] Dependencies are updated to latest versions or new tickets are created if there are breaking changes or deprecations
- [ ] If an unrelated part of the codebase needs to be updated or refactored, create tickets with proposed changes

### CODE REVIEW

**Review by StanevPrime — CHANGES_REQUESTED** [review]
CHANGELOG.md, package versions in pubspec.yaml and https://github.com/Prime-Holding/rx_bloc/blob/develop/packages/rx_bloc_cli/lib/src/rx_bloc_cli_constants.dart#L2 needs to be updated

File: packages/rx_bloc_cli/mason_templates/bricks/lib_router/__brick__/lib/lib_router/router.dart [hunk 1]
```diff
@@ -156,7 +156,7 @@ class AppRouter {
     }
 
     final pathInfo =
-        router.routeInformationParser.configuration.findMatch(state.uri.toString());
+        router.routeInformationParser.configuration.findMatch(state.uri);
```
[the remark this answers is not in the document]
Comment (nikolay-vasilev-prime) [reply packages/rx_bloc_cli/mason_templates/bricks/lib_router/__brick__/lib/lib_router/router.dart:159]: It works
Screenshot 2024-08-27 at 12 55 36

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/test/helpers/goldens_file_comparator.dart [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+class GoldensFileComparator extends LocalFileComparator {
+  static const double _kGoldenDiffTolerance = 0.0041;
+
+  GoldensFileComparator(String testFile)
+      : super(Uri.parse(_getTestFile(testFile)));
+
+  @override
+  Future<bool> compare(Uint8List imageBytes, Uri golden) async {
+    final dsa = Uri.parse(golden.path.split('/').last);
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/test/helpers/goldens_file_comparator.dart:14]: What `dsa` means ?

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/test/helpers/goldens_file_comparator.dart [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import 'dart:developer';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+class GoldensFileComparator extends LocalFileComparator {
+  static const double _kGoldenDiffTolerance = 0.0041;
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/test/helpers/goldens_file_comparator.dart:7]: Why we need `0.0041` tolerance ?

**Review by StanevPrime — CHANGES_REQUESTED** [review]

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/test/helpers/goldens_file_comparator.dart (hunk 3 above)
Comment (DDavidPrime) [reply packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/test/helpers/goldens_file_comparator.dart:7]: @StanevPrime  Copied from the booking app. Any suggestions for the minimal tolerance?

Comment (StanevPrime) [reply packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/test/helpers/goldens_file_comparator.dart:7]: 0, unless is needed more

Comment (DDavidPrime) [reply packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/test/helpers/goldens_file_comparator.dart:7]: While having some small margin for error is always a good thing to have, we can go with `0`. Although judging from previous experience, this has always been a pain point in the CI.

**Review by DDavidPrime — APPROVED** [review]

**Review by StanevPrime — CHANGES_REQUESTED** [review]

**Review by StanevPrime — APPROVED** [review]

**Review by StanevPrime — APPROVED** [review]

**Pull request merged**